books/RayTracingTheRestOfYourLife.html

```diff
@@ -423,184 +571,504 @@
 
-<div class='together'>
-Note that for $x$ where we didn’t define $p(x)$, $p(x) = 0$, _i.e._, the probability of an $x$ there
-is zero. For our example PDF $p(r) = r/2$, the $P(x)$ is:
+Finding the probability of a region surrounding x may not be zero:
+
+    $$ \text{Probability} (r | x - \Delta x < r < x + \Delta x ) =
+         \text{area}(p(r), x - \Delta x, x + \Delta x) $$
+    $$ = P(x + \Delta x) - P(x - \Delta x) $$
 
-  $$ P(x) = 0 : x < 0                 $$
-  $$ P(x) = \frac{x^2}{4} : 0 < x < 2 $$
-  $$ P(x) = 1 : x > 2                 $$
 </div>
 
+Choosing our Samples
+--------------------
+If we have a PDF for the function that we care about, then we have the probability that the function
+will return a value within an arbitrary interval. We can use this to determine where we should
+sample. Remember that this started as a quest to determine the best way to sample a scene so that we
+wouldn't get very bright pixels next to very dark pixels. If we have a PDF for the scene, then we
+can probabilistically steer our samples toward the light without making the image inaccurately
+bright. We already said that if we steer our samples toward the light then we _will_ make the image
+inaccurately bright. We need to figure out how to steer our samples without introducing this
+inaccuracy, this will be explained a little bit later, but for now we'll focus on generating samples
+if we have a PDF. How do we generate a random number with a PDF? For that we will need some more
+machinery. Don’t worry this doesn’t go on forever!
+
 <div class='together'>
-One question is, what’s up with $x$ versus $r$? They are dummy variables -- analogous to the
-function arguments in a program. If we evaluate $P$ at $x = 1.0$, we get:
+Our random number generator `random_double()` produces a random double between 0 and 1. The number
+generator is uniform between 0 and 1, so the double has an equal likelikhood of being any number
+between 0 and 1. If our PDF is uniform over a domain, say $a = 0$ and $b = 10$, then we can
+trivally produce perfect samples for this uniform PDF with
+
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    10.0 * random_double()
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+That's an easy case, but the vast majority of cases that we're going to care about are nonuniform.
+We need to figure out a way to convert a uniform random number generator into a nonuniform random
+number generator, where the distribution is defined by the PDF. We'll just _assume_ that there
+exists a function $f(d)$ that takes uniform input and produces a nonuniform distribution weighted by
+PDF. Now we need to figure out a way to solve for $f(d)$.
+
+For the PDF given above, where $p(r) = \frac{r}{2}$, the probability of a random sample is higher
+toward 2 than it is toward 0. More 1.8s and 1.9s should be generated than 0.1s and 0.2s. If we put
```

Comment:
If I say that the numberd 1.8 and 1.9 have a higher chance than 0.1 and 0.2, then I am saying that a specific number has a discrete probability of happening. This contradicts my hammering of the point above that the probability of any number being generated is zero.

I think I'll just rewrite something like "greater probability of getting a number between 1.8 and 2.0 than 0.0 and 0.2"

Reply:
Yes, I know that _our_ RNG is discrete, but it would muddy the text to mention that point in the text...

Reply:
Actually, I was just rewording what you said, but I see what you mean about specific values. I agree that it might be an improvement to talk about one range having more probability than some other range.